File: src/main/java/org/cloudfoundry/credhub/util/CurrentTimeProvider.java
package org.cloudfoundry.credhub.util;

import java.time.Instant;
import java.time.temporal.TemporalAccessor;
import java.util.Optional;

import org.springframework.data.auditing.DateTimeProvider;
import org.springframework.stereotype.Component;

@Component
public class CurrentTimeProvider implements DateTimeProvider {
  @Override
  public Optional<TemporalAccessor> getNow() {
    return Optional.of(getInstant());
  }

  public Instant getInstant() {
    return Instant.now();
  }

  public long currentTimeMillis() {
    return System.currentTimeMillis();
  }

  public void sleep(long sleepTimeInMillis) throws InterruptedException {
    Thread.sleep(sleepTimeInMillis);
  }
}

File: components/time-support/src/main/java/org/cloudfoundry/credhub/util/CurrentTimeProvider.kt
package org.cloudfoundry.credhub.util

import java.time.Instant
import java.time.temporal.TemporalAccessor
import java.util.Optional

import org.springframework.data.auditing.DateTimeProvider
import org.springframework.stereotype.Component

@Component
class CurrentTimeProvider : DateTimeProvider {
    val instant: Instant
        get() = Instant.now()

    override fun getNow(): Optional<TemporalAccessor> {
        return Optional.of(instant)
    }

    fun currentTimeMillis(): Long {
        return System.currentTimeMillis()
    }

    @Throws(InterruptedException::class)
    fun sleep(sleepTimeInMillis: Long) {
        Thread.sleep(sleepTimeInMillis)
    }
}
